Sources/Workspace/InitPackage.swift

```diff
@@ -45,7 +45,7 @@ public final class InitPackage {
         case executable = "executable"
         case systemModule = "system-module"
         case manifest = "manifest"
-        case `extension` = "extension"
+        case commandPlugin = "command-plugin"
```

Comment:
Since there will likely be new kinds of plugins over time (there is already also `build tool` for example), should this perhaps be a `plugin` case with an associated value?

Reply:
+1 for this, `plugin` with an associated value is also something I'd prefer

Reply:
On a second thought, that could make it harder for `InitPackage.PackageType` to conform to `Codable`, which is required for ArgumentParser for this to be a valid CLI option.

Reply:
Right, I was thinking we still want the CLI options to stay flat (e.g. `--type command-plugin`, `--type build-plugin` etc). But in theory we could also introduce some kind of "subtype", e.g. `--type plugin --subtype command` (not entirely sure how it would need to be modeled for argument parser).

Reply:
I was thinking from just the `InitPackage` API perspective, and so decoupling from the argument parser would make sense to me.  A `--subtype` is interesting, but maybe something best left to a second PR, since that should maybe have some forum discussion if scripts already use it, etc?

@neonichu I'd like to help with this change — would you prefer I add commits to your branch or start a new PR based on your commit?  Maybe better to just continue this PR?

Reply:
Thanks, I'm fine with whatever is easier for you.